cmd/maya-apiserver/app/server/volume_endpoint_v1alpha1.go

```diff
@@ -57,56 +55,196 @@ func (v *volumeAPIOpsV1alpha1) get() (interface{}, error) {
 
 	case strings.Contains(path, "/info/"):
 		volName := strings.TrimPrefix(path, "/info/")
+		// read a cas volume
 		return v.read(volName)
-	case strings.Contains(path, "/delete/"):
-		volName := strings.TrimPrefix(path, "/delete/")
-		return v.delete(volName)
 	case path == "/":
+		// list cas volumes
 		return v.list()
 	default:
 		return nil, CodedError(405, ErrInvalidMethod)
 	}
 }
 
-func (v *volumeAPIOpsV1alpha1) create() (interface{}, error) {
+// httpDelete deals with http delete request method
+func (v *volumeAPIOpsV1alpha1) httpDelete() (interface{}, error) {
+	// Extract info from path after trimming
+	path := strings.TrimPrefix(v.req.URL.Path, "/latest/volumes")
+
+	// check if req has valid delete path
+	if path == v.req.URL.Path {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
 
-	glog.Infof("Received volume add request: Version: 'v1alpha1'")
+	switch {
+	case strings.Contains(path, "/delete/"):
+		volName := strings.TrimPrefix(path, "/delete/")
+		// delete a cas volume
+		return v.delete(volName)
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+}
 
-	vol := &v1.Volume{}
+func (v *volumeAPIOpsV1alpha1) create() (*v1alpha1.CASVolume, error) {
+	glog.Infof("cas template based volume create request was received")
 
-	// unmarshall the request to vol
-	if err := decodeBody(v.req, vol); err != nil {
+	vol := &v1alpha1.CASVolume{}
+	err := decodeBody(v.req, vol)
+	if err != nil {
 		return nil, CodedError(400, err.Error())
 	}
 
-	// Name is expected
-	if vol.Name == "" {
-		return nil, CodedError(400, fmt.Sprintf("Missing volume name '%v'", vol))
+	// volume name is expected
+	if len(vol.Name) == 0 {
```

Comment:
Just a opinion here...
What if someone provide `vol.Name= "     " `. (Notice the spaces)
In that case also, we consider it empty. So proper validation should go following:
`len(strings.TrimSpace(vol.Name)) == 0` 
I faced this kind of validation issue once somewhre i don't remember.
Do we require the `len(strings.TrimSpace(vol.Name)) == 0`  validation or the existing will work fine.
What do you say?